### PR TITLE
Adding the week number parameter

### PR DESCRIPTION
**Pull request by samip5**
I have now added the week number parameter and it uses the `arrow` library as suggested. The other edits that were suggested in #1282 are somewhat implemented like I did organize it better I hope.

Regarding number 3 in the last PR, sorry but dunno about that.

I have tested that the newsletter works with the `week_number` in the body of it.

### CODE REVIEW

File: plexpy/common.py [hunk 1]
```diff
@@ -519,6 +520,7 @@
         'parameters': [
             {'name': 'Server Name', 'type': 'str', 'value': 'server_name', 'description': 'The name of your Plex Server.'},
             {'name': 'Start Date', 'type': 'str', 'value': 'start_date', 'description': 'The start date of the newesletter.'},
+            {'name': 'Week Number', 'type': 'int', 'value': 'week_number', 'description': 'The week number of current week.'},
```
Comment (JonnyWong16) [inline plexpy/common.py]: Move after "End Date".

File: plexpy/newsletters.py [hunk 2]
```diff
@@ -308,13 +308,14 @@ class Newsletter(object):
     _TEMPLATE_MASTER = ''
     _TEMPLATE = ''
 
-    def __init__(self, config=None, email_config=None, start_date=None, end_date=None,
+    def __init__(self, config=None, email_config=None, start_date=None, week_number=None, end_date=None,
```
Comment (JonnyWong16) [inline plexpy/newsletters.py]: Doesn't need to be a class attribute.

File: plexpy/newsletters.py [hunk 3]
```diff
@@ -477,6 +481,7 @@ def _build_params(self):
             'server_name': plexpy.CONFIG.PMS_NAME,
             'start_date': self.start_date.format(date_format),
             'end_date': self.end_date.format(date_format),
+            'week_number': self.week_number,
```
Comment (JonnyWong16) [inline plexpy/newsletters.py]:
```python
'week_number': self.start_date.isocalendar()[1]
```

File: plexpy/newsletters.py [hunk 4]
```diff
@@ -308,13 +308,14 @@ class Newsletter(object):
...
 
-    def __init__(self, config=None, email_config=None, start_date=None, end_date=None,
+    def __init__(self, config=None, email_config=None, start_date=None, week_number=None, end_date=None,
                  subject=None, body=None, message=None):
         self.config = self.set_config(config=config, default=self._DEFAULT_CONFIG)
         self.email_config = self.set_config(config=email_config, default=self._DEFAULT_EMAIL_CONFIG)
         self.uuid = generate_newsletter_uuid()
 
         self.start_date = None
+        self.week_number = None
```
Comment (JonnyWong16) [inline plexpy/newsletters.py]: Remove.

File: plexpy/newsletters.py [hunk 5]
```diff
@@ -332,6 +333,9 @@ def __init__(self, config=None, email_config=None, start_date=None, end_date=Non
             except ValueError:
                 pass
 
+        if week_number is None:
```
Comment (JonnyWong16) [inline plexpy/newsletters.py]: Remove.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]
Changes below.

**Review by samip5 — COMMENTED** [review]
I have now done the suggested edits and updated the branch.

**Review by JonnyWong16 — APPROVED** [review]

**Comment by JonnyWong16** [comment]
Thanks.

**Pull request merged**